pwiz/utility/misc/VendorReaderTestHarness.cpp

```diff
@@ -387,12 +402,15 @@ void testRead(const Reader& reader, const string& rawpath, const bfs::path& pare
         // test SpectrumList::min_level_accepted() for vendors
         if (msd.run.spectrumListPtr && msd.run.spectrumListPtr->size() > 0)
         {
-            DetailLevel msLevelDetailLevel = msd.run.spectrumListPtr->min_level_accepted([](const Spectrum& s) { return s.hasCVParam(MS_ms_level); });
+            DetailLevel msLevelDetailLevel = msd.run.spectrumListPtr->min_level_accepted([](const Spectrum& s) { return !ContainsMsInfo(s)
```

Comment:
Good catch, I didn't realize we didn't have any existing tests with non-MS spectra which would have already triggered this edge case. I think we can simplify the predicate though and avoid the need for `ContainsMsInfo`:
```
DetailLevel msLevelDetailLevel = msd.run.spectrumListPtr->min_level_accepted([](const Spectrum& s) { return s.hasCVParamChild(MS_mass_spectrum) ? s.hasCVParam(MS_ms_level) : boost::indeterminate;});
```
(indeterminate means it needs to try another spectrum, false means it will try a higher detail level; the latter would be wrong for an EMR spectrum because even the highest detail level won't have an MS level)

If the spectrumList doesn't have ANY MS spectra, then this would still fail. But that's not the case for your example file, right?

Reply:
Thank you very much. Okay, I'll take a look at that. I'm not sure if there is another Ms spectrum in my sample. I will try to find out if there is one.

Reply:
My idea was to extend the list for other spectra which don´t have a ms spectrum. But, sure, your solution is more general. But I still have to check if my example has other ms spectra.

Reply:
I took another closer look at the code and I think I understand it better now. My example would find a Ms1 Spectrum, even if none would be present if we had an EMR Spectrum. That would certainly not be correct. The way you suggested it should work if there was at least 1 Ms1 Spectrum. Seems to work now. I´ll check it in.